File: task-03/Python/Sub-Task 4/subtask4.py
with open('input.txt', 'r') as infile:
    n = int(infile.read().strip())
with open('output.txt', 'w') as outfile:
    for i in range(n):
        outfile.write(" " * (n - i - 1) + "*" * (2 * i + 1) + "\n")
    for i in range(n - 2, -1, -1):
        outfile.write(" " * (n - i - 1) + "*" * (2 * i + 1) + "\n")

File: task-03/Ruby/Sub-Task 4/subtask4.rb
n = File.read('input.txt').to_i
File.open('output.txt', 'w') do |file|
  (0...n).each { |i| file.puts " " * (n - i - 1) + "*" * (2 * i + 1) }
  (n-2).downto(0) { |i| file.puts " " * (n - i - 1) + "*" * (2 * i + 1) }
end
